sympy/solvers/solvers.py

```diff
@@ -2685,6 +2742,7 @@ def _tsolve(eq, sym, **flags):
             elif lhs.func == LambertW:
                 return _solve(lhs.args[0] - rhs*exp(rhs), sym, **flags)
 
+        # XXX this is done above...will it be necessary here in any case?
```

Comment:
In this case it is required. `solve((x/2)**(2/x) - sqrt(2))`

Reply:
I reorganized things a bit and did both the logform and rewrite; this example you give is a power and should have been solved in the power handling portion of the code

Reply:
The thing is rewrite is needed in many different cases. You can see in travis. I think it would be improper to remove it. It is not only related to power. Actually many multiplication equations are solved with rewrite.

Reply:
Cases like this also needs rewrite `solve(sin(x) + tan(x)) == [0, -pi, pi, 2*pi]`